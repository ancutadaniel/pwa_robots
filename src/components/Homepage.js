import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import Header from '../components/Header';
import './Homepage.css';

class Homepage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      color1: '',
      color2: '',
    };
    this.setColor = this.setColor.bind(this);
    this.setColor1 = this.setColor1.bind(this);
  }

  componentDidMount() {
    this.props.onRequestRobots();
  }

  setColor(event) {
    this.setState({ color1: event.target.value });
  }

  setColor1(event) {
    this.setState({ color2: event.target.value });
  }

  filteredRobots = () => {
    return this.props.robots.filter((robot) => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return isPending ? (
      <h1>Loading</h1>
    ) : (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <label></label>
        <input
          id='color1'
          type='color'
          name='color1'
          onChange={this.setColor}
          value='#ffffff'
        />
        <label></label>
        <input
          id='color2'
          type='color'
          name='color1'
          onChange={this.setColor1}
          value='#ffffff'
        />
        <Scroll color1={this.state.color1} color2={this.state.color2}>
          <CardList robots={this.filteredRobots()} />
        </Scroll>
      </div>
    );
  }
}

export default Homepage;
