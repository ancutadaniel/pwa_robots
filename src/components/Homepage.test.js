import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Homepage from './Homepage';

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    search: '',
    isPending: false,
  };
  wrapper = shallow(<Homepage {...mockProps} />);
});

it('expect to render Homepage component', () => {
  expect(wrapper).toMatchSnapshot();
});

it('filter robots correctly', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 2,
        name: 'John',
        email: 'john@gmail.com',
      },
    ],
    searchField: 'john',
    isPending: false,
  };
  const wrapper2 = shallow(<Homepage {...mockProps2} />);

  expect(wrapper2.instance().filteredRobots()).toEqual([
    {
      id: 2,
      name: 'John',
      email: 'john@gmail.com',
    },
  ]);
});

it('filter robots correctly 2', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 2,
        name: 'John',
        email: 'john@gmail.com',
      },
    ],
    searchField: 'a',
    isPending: false,
  };
  const filteredRobots = [];
  const wrapper3 = shallow(<Homepage {...mockProps3} />);

  expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots);
});

it('input colors', () => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 2,
        name: 'John',
        email: 'john@gmail.com',
      },
    ],
    searchField: 'a',
    isPending: false,
  };
  const setColor = jest.fn();
  const wrapper = mount(<Homepage {...mockProps} />);
  expect(wrapper.state()).toEqual({ color1: '', color2: '' });
});
