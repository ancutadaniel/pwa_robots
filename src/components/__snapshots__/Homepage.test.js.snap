// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Homepage component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <label />
  <input
    id="color1"
    name="color1"
    onChange={[Function]}
    type="color"
    value="#ffffff"
  />
  <label />
  <input
    id="color2"
    name="color1"
    onChange={[Function]}
    type="color"
    value="#ffffff"
  />
  <Scroll
    color1=""
    color2=""
  >
    <CardList
      robots={Array []}
    />
  </Scroll>
</div>
`;
